pandas/io/common.py

```diff
@@ -329,27 +355,31 @@ def _get_handle(path_or_buf, mode, encoding=None, compression=None,
     path_or_buf = _stringify_path(path_or_buf)
     is_path = isinstance(path_or_buf, str)
 
+    compression_method = None
     if is_path:
-        compression = _infer_compression(path_or_buf, compression)
+        compression_method = _infer_compression(path_or_buf, compression)
 
-    if compression:
+    if compression_method:
 
         # GZ Compression
-        if compression == 'gzip':
+        if compression_method == 'gzip':
             if is_path:
                 f = gzip.open(path_or_buf, mode)
             else:
                 f = gzip.GzipFile(fileobj=path_or_buf)
 
         # BZ Compression
-        elif compression == 'bz2':
+        elif compression_method == 'bz2':
             if is_path:
                 f = bz2.BZ2File(path_or_buf, mode)
             else:
                 f = bz2.BZ2File(path_or_buf)
 
         # ZIP Compression
-        elif compression == 'zip':
+        elif compression_method == 'zip':
+            arcname = None
+            if isinstance(compression, dict) and 'arcname' in compression:
+                arcname = compression['arcname']
```

Comment:
For greater flexibility, pop out `method` and pass in `compression` as `kwargs`.

Reply:
Good idea; To clarify, do you mean to add an additional parameter to `ByteZipFile.__init__` which specifies kwargs for `ByteZipFile.write`'s super call to `ZipFile.writestr`? I originally went with passing `arcname` to `ByteZipFile.__init__` as it may be wrapped in a `TextIOWrapper`, which does not support passing kwargs in `write` to the wrapped `ByteZipFile`.  `ZipFile` itself requires that the archived file name be passed in `write`, so I don't think passing it as `kwargs` to `ByteZipFile` would work out.

Reply:
Something like this:

~~~
compression.pop("method")
zf = BytesZipFile(path_or_buf, mode, **compression)
~~~